### PR TITLE
Handling 503 errors from Slack

### PR DESCRIPTION
**Pull request by lasryaric**
This is my take on handling error from the Slack SDK. 
It uses a Javascript Proxy to catch all exception thrown by the SDK and send the ones we want as "dust handled errors".

Let me know what you think.

### CODE REVIEW

**Review by spolu — DISMISSED** [review]
Looks great!

The proxy pattern looks neat. What is the core distinctive value vs try catch? If I understand correctly the fact that you write the handler only once right?

**Comment by lasryaric** [comment]
Yes. The caller does not need to worry about that at all.

**Comment by lasryaric** [comment]
And the recursive proxy is because Slack has sub object in its sdk (client.conversations.list()). Took me some time to get it right 😅

**Comment by spolu** [comment]
@lasryaric hmmm there seems to be an alignment problem of the connect button on the managed data sources in your screenshot no? (Which I don’t see on current main)

is that your commit or mine?

**Comment by lasryaric** [comment]
There is a little bug to fix in the recursive proxy of the Slack client in this PR before merging it. I will take care of it asap.

**Comment by spolu** [comment]
@lasryaric what's the status here?

**Comment by lasryaric** [comment]
@spolu I added a commit. Ready to merge. I will wait for your review or merge tomorrow morning.
I re-tested everything.

**Review by spolu — APPROVED** [review]

**Pull request merged**
